### PR TITLE
Backport of GCS refactoring

### PR DESCRIPTION
**Pull request by nicholaspcr**
#### Summary

OS counterpart of https://github.com/TheThingsIndustries/lorawan-stack/pull/4310

#### Changes

- Deprecate DCS `authorizeGateway` and `UnauthorizeGateway`
- Remove `authorize` and `unauthorize` CLI sub-commands
- Update tests everywhere and improve code coverage.
- Introduce clean gateway claiming flow that only creates gateways.

##### Steps

For this OS PR the only tests performed are the unite tests, no manual testing yet.

#### Notes for Reviewers

This is a backport of a PR done in https://github.com/TheThingsIndustries/lorawan-stack/pull/4310

I went over each file, double checking the contents but its worth having a second read by someone else.

@KrishnaIyer this backport was done by @johanstokking's request in the hopes it alleviates your workload. 

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Testing: The steps/process to test this feature are clearly explained including testing for regressions.
- [ ] Infrastructure: If infrastructural changes (e.g., new RPC, configuration) are needed, a separate issue is created in the infrastructural repositories.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by KrishnaIyer** [comment]
Thanks I'll pick this (for review I mean)

File: api/ttn/lorawan/v3/deviceclaimingserver.proto [hunk 1]
```diff
@@ -286,6 +286,7 @@ service EndDeviceBatchClaimingServer {
   }
 }
 
+// DEPRECATED: This message is deprecated and will be removed in a future version of The Things Stack.
 message CUPSRedirection {
```
Comment (KrishnaIyer) [inline api/ttn/lorawan/v3/deviceclaimingserver.proto:290]: Please add `option deprecated = true` here as well as the other messages.

**Review by KrishnaIyer — CHANGES_REQUESTED** [review]
LGTM with a few comments.
- Please also backport the observability changes for unclaiming.

**Review by KrishnaIyer — APPROVED** [review]
LGTM. Please fix CI, merge and do a merge to enterprise.

**Review by johanstokking — APPROVED** [review]

**Pull request merged**